// Copyright (c) 2020-2021 The Open-Transactions developers
// This Source Code Form is subject to the terms of the Mozilla Public
// License, v. 2.0. If a copy of the MPL was not distributed with this
// file, You can obtain one at http://mozilla.org/MPL/2.0/.

syntax = "proto2";

package opentxs.proto;
option java_package = "org.opentransactions.proto";
option java_outer_classname = "OTAuthority";
option optimize_for = LITE_RUNTIME;

import public "Enums.proto";
import public "Credential.proto";

message Authority
{
    optional uint32 version = 1;
    optional string nymID = 2;
    optional string masterID = 3;
    optional AuthorityMode mode = 4;
    optional uint32 index = 5;
    optional Credential masterCredential = 6;
    repeated string activeChildIDs = 7;
    repeated Credential activeChildren = 8;
    repeated string revokedChildIDs = 9;
    repeated Credential revokedChildren = 10;
}
